src/game/savegame/savegame_bson.c

```diff
@@ -1219,6 +1266,13 @@ bool Savegame_BSON_LoadFromFile(MYFILE *fp, GAME_INFO *game_info)
         goto cleanup;
     }
 
+    if (header.version >= VERSION_3) {
+        if (!SaveGame_BSON_LoadCurrentMusic(
+                json_object_get_object(root_obj, "music"))) {
+            goto cleanup;
+        }
+    }
+
```

Comment:
I don't think it's necessary to bump the version, as your scheme is forwards compatible?  I don't remember our SG versioning policy. Anyway we could extract the JSON retrieval to a variable, `music_obj`.

Reply:
Hm maybe. I don't need it for this though? I guess the game just warns it's not found and continues loading. I'll leave the decision up to you.

```
    if (header.version >= VERSION_3) {
        if (!SaveGame_BSON_LoadCurrentMusic(
                json_object_get_object(root_obj, "music"))) {
            goto cleanup;
        }
    }
```

Also not sure what you mean by ` Anyway we could extract the JSON retrieval to a variable, music_obj`. I already have a `json_object_s music_obj`.

Reply:
Something like this:

```diff
     if (header.version >= VERSION_3) {
-        if (!SaveGame_BSON_LoadCurrentMusic(
-                json_object_get_object(root_obj, "music"))) {
+        json_object_s music_obj = json_object_get_object(root_obj, "music");
+        if (!SaveGame_BSON_LoadCurrentMusic(music_obj)) {
             goto cleanup;
         }
     }
```

As to whether to bump the sg version, I'm indifferent, maybe @lahm86 has stronger opinion on this?

Reply:
> ```diff
> "
> ```

Ok I can do it this way. No other loading call does it this way FYI. What does it accomplish?

Reply:
My thinking was to reduce the indentation a bit, but I'd say consistency here is more important than looks, so it's OK to leave it as-is if it follows an established way.